### DISCUSSION TITLE
能否支持JSRuntime创建多个JSContext上下文？

### DISCUSSION THREAD

**Discussion by jbactb** [HarlonWang/quickjs-wrapper#113] [Q&A] [answered]
能否支持JSRuntime运行时 创建多个JSContext上下文？

如果只有一个JSContext的话加载多个脚本存在同名属性方法，会导致全局GlobalObjec冲突。

我看quickjs文档是支持一个JSRuntime多个JSContext，每个JSContext都有自己的全局对象和系统对象。

**Comment by HarlonWang** [comment, marked as answer]
目前还没有明确的计划去支持，另外，即使是一个 JSRuntime，多个 JSContext 之前也是相互隔离的，所以如果是解决存在同名属性方法的问题，现在你也可以直接创建多个 JSContext 的

**Comment by jbactb** [comment]
就是实现数据隔离，现在QuickJSContext.create()会同时创建JSRuntime和JSContext，
如果太多JSRuntime内存占可能有点多

**Comment by HarlonWang** [comment]
嗯，后面可能会考虑分离 JSRuntime 和 JSContext 的创建逻辑，还没有具体计划，你可以 clone 下来直接修改，也比较简单的，不过，JSRuntime 内存占用不多的，主要就是一些静态类的初始化信息，你也可以通过调用 dumpMemeory 看下内存占用情况

**Comment by jbactb** [comment]
ok 谢谢
